multiuser/keycloak/che-multiuser-keycloak-server/src/main/java/org/eclipse/che/multiuser/keycloak/server/KeycloakEnvironmentInitalizationFilter.java

```diff
@@ -37,135 +36,146 @@
 import org.eclipse.che.commons.env.EnvironmentContext;
 import org.eclipse.che.commons.subject.Subject;
 import org.eclipse.che.commons.subject.SubjectImpl;
-import org.eclipse.che.commons.tracing.TracingTags;
+import org.eclipse.che.multiuser.api.authentication.commons.SessionStore;
+import org.eclipse.che.multiuser.api.authentication.commons.filter.MultiuserEnvironmentInitializationFilter;
 import org.eclipse.che.multiuser.api.permission.server.AuthorizedSubject;
 import org.eclipse.che.multiuser.api.permission.server.PermissionChecker;
 import org.eclipse.che.multiuser.keycloak.shared.KeycloakConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Sets subject attribute into session based on keycloak authentication data.
  *
  * @author Max Shaposhnik (mshaposhnik@redhat.com)
  */
 @Singleton
-public class KeycloakEnvironmentInitalizationFilter extends AbstractKeycloakFilter {
+public class KeycloakEnvironmentInitalizationFilter
+    extends MultiuserEnvironmentInitializationFilter {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(KeycloakEnvironmentInitalizationFilter.class);
 
   private final KeycloakUserManager userManager;
+  private final KeycloakProfileRetriever keycloakProfileRetriever;
   private final RequestTokenExtractor tokenExtractor;
   private final PermissionChecker permissionChecker;
   private final KeycloakSettings keycloakSettings;
-  private final KeycloakProfileRetriever keycloakProfileRetriever;
-  private final Tracer tracer;
+  private final JwtParser jwtParser;
 
   @Inject
   public KeycloakEnvironmentInitalizationFilter(
+      SessionStore sessionStore,
+      JwtParser jwtParser,
       KeycloakUserManager userManager,
       KeycloakProfileRetriever keycloakProfileRetriever,
       RequestTokenExtractor tokenExtractor,
       PermissionChecker permissionChecker,
       KeycloakSettings settings,
       Tracer tracer) {
+    super(sessionStore, tokenExtractor);
+    this.jwtParser = jwtParser;
     this.userManager = userManager;
+    this.keycloakProfileRetriever = keycloakProfileRetriever;
     this.tokenExtractor = tokenExtractor;
     this.permissionChecker = permissionChecker;
     this.keycloakSettings = settings;
-    this.keycloakProfileRetriever = keycloakProfileRetriever;
-    this.tracer = tracer;
   }
 
+  @Override
+  public void init(FilterConfig filterConfig) {}
+
   @Override
   public void doFilter(ServletRequest request, ServletResponse response, FilterChain filterChain)
       throws IOException, ServletException {
 
-    final HttpServletRequest httpRequest = (HttpServletRequest) request;
-    final String token = tokenExtractor.getToken(httpRequest);
+    final String token = tokenExtractor.getToken((HttpServletRequest) request);
+    if (token == null) {
+      sendError(response, 401, "Authorization token is missed");
```

Comment:
can we please also log all errors?

Reply:
Not sure if we should pollute log with just incorrect user input data.